packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/EmailPasswordAuthTests.kt

```diff
@@ -152,45 +153,50 @@ class EmailPasswordAuthWithAutoConfirmTests {
-    //            val user = app.login(Credentials.emailPassword(email, "new-password"))
-    //            user.logOut()
-    //        } finally {
-    //            adminApi.setResetFunction(enabled = false)
-    //        }
-    //    }
-    // }
-    //
-    // @Test
-    // fun callResetPasswordFunction_invalidServerArgsThrows() {
-    //    val provider = app.emailPassword
-    //    admin.setResetFunction(enabled = true)
-    //    val email = TestHelper.getRandomEmail()
-    //    provider.registerUser(email, "123456")
-    //    try {
-    //        provider.callResetPasswordFunction(email, "new-password", "wrong-magic-word")
-    //    } catch (error: AppException) {
-    //        assertEquals(ErrorCode.SERVICE_UNKNOWN, error.errorCode)
-    //    } finally {
-    //        admin.setResetFunction(enabled = false)
-    //    }
-    // }
-    //
-    // @Test
-    // fun callResetPasswordFunction_invalidArgumentsThrows() {
-    //    val provider = app.emailPassword
-    //    assertFailsWith<IllegalArgumentException> { provider.callResetPasswordFunction(TestHelper.getNull(), "password") }
-    //    assertFailsWith<IllegalArgumentException> { provider.callResetPasswordFunction("foo@bar.baz", TestHelper.getNull()) }
-    // }
+    @Test
+    fun callResetPasswordFunction() {
+        val provider = app.emailPasswordAuth
+        val adminApi = app.asTestApp
+        runBlocking {
+            adminApi.setResetFunction(enabled = true)
+            val email = TestHelper.randomEmail()
+            provider.registerUser(email, "123456")
+            try {
+                provider.callResetPasswordFunction(email, "new-password", "say-the-magic-word", 42)
+                val user = app.login(Credentials.emailPassword(email, "new-password"))
+                user.logOut()
+            } finally {
+                adminApi.setResetFunction(enabled = false)
+            }
+        }
+    }
+
+    @Test
+    fun callResetPasswordFunction_invalidServerArgsThrows() {
+        val provider = app.emailPasswordAuth
+        val adminApi = app.asTestApp
+        runBlocking {
+            adminApi.setResetFunction(enabled = true)
+            val email = TestHelper.randomEmail()
+            provider.registerUser(email, "123456")
+            try {
+                provider.callResetPasswordFunction(email, "new-password", "wrong-magic-word")
+            } catch (error: AppException) {
+                assertTrue(error.message!!.contains("failed to reset password for user \"$email\""), error.message)
+            } finally {
+                adminApi.setResetFunction(enabled = false)
+            }
+        }
+    }
+
+    @Test
+    fun callResetPasswordFunction_invalidArgumentsThrows() {
+        val provider = app.emailPasswordAuth
+        runBlocking {
+            assertFailsWith<IllegalArgumentException> { provider.callResetPasswordFunction("", "password") }
```

Comment:
I assume you cannot use nulls here at runtime so this is the closest invalid argument I came up with based on old tests, not sure if there is more I should check here.

Reply:
That is correct, null pointer errors are not allowed here anymore. But instead we have a new class of errors. If we accept `Any?` as the argument type, we can test here what happens if you use a type that cannot be turned into a `BsonValue`